libbeat/autodiscover/providers/docker/config.go

```diff
@@ -36,15 +39,22 @@ type Config struct {
 
 func defaultConfig() *Config {
 	return &Config{
-		Host:   "unix:///var/run/docker.sock",
-		Prefix: "co.elastic",
+		Host:      "unix:///var/run/docker.sock",
+		Prefix:    "co.elastic",
+		Separator: "/",
 	}
 }
 
 // Validate ensures correctness of config
-func (c *Config) Validate() {
+func (c *Config) Validate() error {
 	// Make sure that prefix doesn't ends with a '.'
 	if c.Prefix[len(c.Prefix)-1] == '.' && c.Prefix != "." {
 		c.Prefix = c.Prefix[:len(c.Prefix)-2]
 	}
+	switch c.Separator {
+	case "/", "-", ".", "_":
```

Comment:
Maybe we should limit the options to `/`, and `.`, any advantage on accepting more values?